### PR TITLE
Preserve maxLength of EmojiEditText

### PR DESCRIPTION
**Pull request by FeuRenard**
### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4 (CyanogenMod)
 * EDIT, 31 Aug 2016: AVD Nexus 5X, Android 6.0
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

----------

### Description
Improves #3438 (further explanation: https://github.com/WhisperSystems/Signal-Android/issues/3388#issuecomment-243606571)

Addresses #3388
// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public EmojiEditText(Context context, AttributeSet attrs) {
   public EmojiEditText(Context context, AttributeSet attrs, int defStyleAttr) {
     super(context, attrs, defStyleAttr);
     if (!TextSecurePreferences.isSystemEmojiPreferred(getContext())) {
-      setFilters(new InputFilter[]{ new EmojiFilter(this) });
+      setFilters(new InputFilter[]{ new EmojiFilter(this), this.getFilters()[0] });
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java]: Is it not possible for getFilters to return an empty array?

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java]: Theoretically yes, but I did not observe this case. I tested with the two `EmojiEditText`s that are used in Signal: conversation input and group update. `this.getFilters().length` always returned 1.

**Comment by FeuRenard** [comment]
I updated the PR with a private method that safely adds the `EmojiFilter` to an array of arbitrary length.

**Comment by moxie0** [comment]
in 3.19.0

**Pull request closed without merging**